### ISSUE TITLE
implement an internal database to improve the user experience

### ISSUE THREAD

**Issue by cpg** [amahi/android#315]
We have talked about a mini database for keeping track of various features, like last media play states, maybe some caching of API returns, etc.

Any time we introduce state we have to be careful about managing it. Here are some DB interface requirements:

  * **resilient to corruption**: it _must not_ (repeat _must not_) ever be corrupted. hardware errors or even bugs could cause this, but it definitely should be very very very rare an not self-inflicted by our code
  * **self-stabilizing**: if it ever does get corruption, it either has to automatically be detected an fixed quickly and swiftly, or it has to be re-initialized quickly so has it works well.
 * **versatile**: has to be used for multiple purposes/features
 * **upgradeable**: it has to be able to be upgradeable automatically, migrating from one schema to the next and with proper data migration. The right hooks have to be in the app to upgrade (and downgrade?) if new uses are needed or formats changed 
 * **transparent**: it must not affect app performance.
 * **consistent**: it has to be consistent with the API data, if used for caching or other purposes, we cannot allow to be inconsistent

**Comment by sanjitchak** [comment]
With an internal database, we may add an Option of checking **recently watched videos**.
We may add an option to add the location of favorite videos and pics to **favorite** section, in internal db.

**Comment by cpg** [comment]
@sanjitchak No, not yet, please. There is another effort for this around a specific feature in #313 that needs to be reviewed first.

**Comment by sanjitchak** [comment]
@cpg  oh! yes. Sorry, I missed it. Thanks, for informing. I will try that out.
